Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx

```diff
@@ -139,9 +153,17 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
         [ currentProblem ],
     );
 
-    const updateSubmissionCode = (code: string | Blob) => {
-        setSubmissionCode(code);
-    };
+    const updateSubmissionCode = useCallback(
+        (code: string | File) => {
+            const { id: problemId } = currentProblem || {};
+            if (isNil(problemId)) {
+                return;
+            }
+            problemSubmissionCode[problemId] = code;
```

Comment:
Mutating state like this is not acceptable in react
You should use something like
```

setProblemSubmissionCode({
     ...problemSubmissionCode,
     problemId: {
         code
     }
```

Reply:
Done.